### PR TITLE
Move comint to a network daemon

### PR DESCRIPTION
**Pull request by elken**
Refactor hacky comint implementation with a more robust and stable
version around the tcp version of the flutter daemon.

Currently debugging doesn't work and there's a slight hang on the initial buffer set-up (this can be offloaded asynchronously) but it has for the most part been supplanted. 

Raising this PR now to get feedback and assess the feasibility of such an implementation.

### CODE REVIEW

**Comment by ericdallo** [comment]
@elken it's important to check since what flutter version TCP was included

File: lsp-dart-flutter-daemon.el [hunk 1]
```diff
@@ -35,6 +34,11 @@
 (defvar lsp-dart-flutter-daemon-commands '())
 (defvar lsp-dart-flutter-daemon-device-added-listeners '())
 
+(defcustom lsp-dart-flutter-daemon--port 9290
```
Comment (ericdallo) [inline lsp-dart-flutter-daemon.el:37]:
```suggestion
(defcustom lsp-dart-flutter-daemon-port 9290
```

File: lsp-dart-flutter-daemon.el [hunk 2]
```diff
@@ -93,26 +105,38 @@ PARAMS is the optional method params."
...
 Call CALLBACK if provided when the receive a response with the built id
 of this command."
   (unless (lsp-dart-flutter-daemon--running-p)
     (lsp-dart-flutter-daemon-start))
   (let* ((id (lsp-dart-flutter-daemon--generate-command-id))
-         (command (lsp-dart-flutter-daemon--build-command id method params)))
-    (add-to-list 'lsp-dart-flutter-daemon-commands (cons id (list :callback callback)))
-    (comint-send-string (get-buffer-process lsp-dart-flutter-daemon-buffer-name) command)))
+         (command (lsp-dart-flutter-daemon--build-command id method params))
+         (client (open-network-stream
```
Comment (ericdallo) [inline lsp-dart-flutter-daemon.el:129]: I think we should handle connection errors similar to how `dap--open-network-stream` does, right?

**Comment by elken** [comment]
JSONRPC seems to be working now, so this is on hold for now and will likely be closed as it's much easier to handle and more async.

**Comment by elken** [comment]
Closed in favor of #165

**Pull request closed without merging**